File: app/src/main/java/com/wojdor/hearthcards/application/update/UpdateResultReceiver.java
package com.wojdor.hearthcards.application.update;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.os.Handler;
import android.support.v4.os.ResultReceiver;

public class UpdateResultReceiver extends ResultReceiver {

    public static final int RESULT_SUCCESS = 1;
    public static final int RESULT_ERROR = 0;

    private Receiver receiver;

    @SuppressLint("RestrictedApi")
    public UpdateResultReceiver(Handler handler) {
        super(handler);
    }

    public void setReceiver(Receiver receiver) {
        this.receiver = receiver;
    }

    @SuppressLint("RestrictedApi")
    @Override
    protected void onReceiveResult(int resultCode, Bundle resultData) {
        if (receiver == null) return;
        receiver.onReceiveResult(resultCode, resultData);
    }

    public interface Receiver {
        void onReceiveResult(int resultCode, Bundle resultData);
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/application/update/UpdateResultReceiver.kt
package com.wojdor.hearthstonecards.application.update

import android.annotation.SuppressLint
import android.os.Bundle
import android.os.Handler
import android.support.v4.os.ResultReceiver

class UpdateResultReceiver @SuppressLint("RestrictedApi")
constructor(handler: Handler) : ResultReceiver(handler) {

    private var receiver: Receiver? = null

    fun setReceiver(receiver: Receiver?) {
        this.receiver = receiver
    }

    @SuppressLint("RestrictedApi")
    override fun onReceiveResult(resultCode: Int, resultData: Bundle?) {
        receiver?.onReceiveResult(resultCode, resultData)
    }

    interface Receiver {
        fun onReceiveResult(resultCode: Int, resultData: Bundle?)
    }

    companion object {
        const val RESULT_SUCCESS = 1
        const val RESULT_ERROR = 0
    }
}
